### PR TITLE
Added test server coverage

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Implemented various tests that were missing.
Removed some superfluous code in xmlgroup_test.go
Removed other_mods.go now that CI has been fixed.

### CODE REVIEW

File: test/autorest/errorsgroup/errorsgroup_test.go [hunk 1]
```diff
@@ -104,7 +105,12 @@ func TestGetPetByIDSuccess1(t *testing.T) {
 }
 
 func TestGetPetByIDSuccess2(t *testing.T) {
-	t.Skip("multiple responses with different schemas NYI")
+	client := getClient(t)
+	result, err := client.GetPetByID(context.Background(), "django")
```
Comment (catalinaperalta) [inline test/autorest/errorsgroup/errorsgroup_test.go:109]: If this is a Get should there also be a comparison of what we actually got? (If we are receiving something)

Comment (jhendrixMSFT) [reply test/autorest/errorsgroup/errorsgroup_test.go:109]: The response for this test only returns a 200 (no response body or headers).

Comment (jhendrixMSFT) [reply test/autorest/errorsgroup/errorsgroup_test.go:109]: Wait let me look into this a bit more....

Comment (jhendrixMSFT) [reply test/autorest/errorsgroup/errorsgroup_test.go:109]: Per the swagger, for an HTTP 200 response you get back a Pet schema.  For an HTTP 202 response there is no schema.

Comment (catalinaperalta) [reply test/autorest/errorsgroup/errorsgroup_test.go:109]: Ah ok, that sounds good then! Is the 202 the final state though for this test?

Comment (catalinaperalta) [reply test/autorest/errorsgroup/errorsgroup_test.go:109]: Well yeah it should be, nvm. Looks good!

**Pull request merged**